bundles/org.eclipse.swt/Eclipse SWT/gtk/org/eclipse/swt/widgets/DateTime.java

```diff
@@ -1831,6 +1831,9 @@ boolean onNumberKeyInput(int key) {
 		if (!Character.isAlphabetic(newText) && !Character.isDigit(newText)) {
 			return false;
 		}
+		if (fieldName == Calendar.AM_PM && (newText != 'p' && newText !='a')){
```

Comment:
Upper case A and P do nothing and even thought what if other locales have different indicators for AM/PM ? As per https://en.wikipedia.org/wiki/12-hour_clock#Abbreviations : The equivalents in [Greek](https://en.wikipedia.org/wiki/Greek_language) are π.μ. and μ.μ., respectively, and in [Sinhala](https://en.wikipedia.org/wiki/Sinhala_language) පෙ.ව. (pe.va.) for පෙරවරු (peravaru, පෙර pera – fore, pre) and ප.ව. (pa.va.) for පස්වරු (pasvaru, පස්සේ passē – after, post). So any such filtering should really not happen

Reply:
Localization support right, okay I'll remove this.